plugin/trino-opa/src/test/java/io/trino/plugin/opa/TestHelpers.java

```diff
@@ -42,59 +44,60 @@ public static InstrumentedHttpClient createMockHttpClient(URI expectedUri, Funct
     {
-        return (OpaAccessControl) OpaAccessControlFactory.create(
-                ImmutableMap.<String, String>builder()
-                        .put("opa.policy.uri", opaUri.toString())
-                        .put("opa.policy.batched-uri", opaBatchUri.toString())
-                        .buildOrThrow(),
-                Optional.of(mockHttpClient),
-                Optional.of(SYSTEM_ACCESS_CONTROL_CONTEXT));
+        runIllegalResponseTestCases(methodToTest, opaConfig, expectedUri);
     }
 
-    public static void assertAccessControlMethodThrowsForIllegalResponses(Consumer<OpaAccessControl> methodToTest)
-    {
-        runIllegalResponseTestCases(methodToTest, TestHelpers::buildAuthorizerWithPredefinedResponse);
-    }
-
-    public static void assertBatchAccessControlMethodThrowsForIllegalResponses(Consumer<OpaAccessControl> methodToTest)
-    {
-        runIllegalResponseTestCases(methodToTest, TestHelpers::buildBatchAuthorizerWithPredefinedResponse);
-    }
-
-    public static OpaAccessControl buildBatchAuthorizerWithPredefinedResponse(MockResponse response)
-    {
-        return createOpaAuthorizer(OPA_SERVER_URI, OPA_SERVER_BATCH_URI, createMockHttpClient(OPA_SERVER_BATCH_URI, request -> response));
-    }
-
-    public static void assertAccessControlMethodThrows(
-            Runnable methodToTest,
+    public static void assertAccessControlMethodThrowsForResponse(
+            MockResponse response,
+            URI expectedUri,
+            OpaConfig opaConfig,
+            Consumer<OpaAccessControl> methodToTest,
             Class<? extends OpaQueryException> expectedException,
             String expectedErrorMessage)
     {
-        assertThatThrownBy(methodToTest::run)
+        InstrumentedHttpClient httpClient = createMockHttpClient(expectedUri, request -> response);
+        assertThatThrownBy(() -> methodToTest.accept(createOpaAuthorizer(opaConfig, httpClient)))
                 .isInstanceOf(expectedException)
                 .hasMessageContaining(expectedErrorMessage);
+        assertThat(httpClient.getRequests()).isNotEmpty();
     }
 
-    private static void runIllegalResponseTestCases(
-            Consumer<OpaAccessControl> methodToTest,
-            Function<MockResponse, OpaAccessControl> authorizerBuilder)
+    public static Map<String, String> opaConfigToDict(OpaConfig config)
+    {
+        ConfigurationMetadata<OpaConfig> metadata = ConfigurationMetadata.getValidConfigurationMetadata(OpaConfig.class);
+        ImmutableMap.Builder<String, String> opaConfigBuilder = ImmutableMap.builder();
+        try {
+            for (ConfigurationMetadata.AttributeMetadata attribute : metadata.getAttributes().values()) {
+                convertPropertyToString(attribute.getGetter().invoke(config)).ifPresent(
+                        propertyValue -> opaConfigBuilder.put(attribute.getInjectionPoint().getProperty(), propertyValue));
+            }
+        }
```

Comment:
Fair point. I was conflicted about whether I liked this or not, I was only trying to avoid having to keep duplicates of every config key.

I'll change this function

Reply:
As a side note, do you think there's any merit to trying to contribute a function that does this to Airlift? Or is it intentionally not provided?

Reply:
Airlift supports configurations that can't be reversed like this